packages/cozy-client/src/models/cron.js

```diff
@@ -0,0 +1,69 @@
+// @ts-check
+import { randomDayTime } from './daytime'
+
+const DAILY = 'daily'
+const HOURLY = 'hourly'
+const MONTHLY = 'monthly'
+const WEEKLY = 'weekly'
+const VALID_FREQUENCIES = [DAILY, HOURLY, MONTHLY, WEEKLY]
+
+const DEFAULT_FREQUENCY = WEEKLY
+// By default konnectors are run at random hour between 12:00PM and 05:00AM.
+const DEFAULT_TIME_INTERVAL = [0, 5]
+
+/**
+ * @typedef {object} FrequencyOptions - frequency options object
+ * @property {Number} [dayOfMonth] - day of the month
+ * @property {Number} [dayOfWeek] - day of the week
+ * @property {Number} [hours] - hours
+ * @property {Number} [minutes] - minutes
+ */
+
+/**
+ * Build a cron string for given frequency with given options
+ * See https://docs.cozy.io/en/cozy-stack/jobs/#cron-syntax
+ *
+ * @param  {'hourly'|'daily'|'weekly'|'monthly'} frequency Frequency
+ * @param  {FrequencyOptions} [options] - FrequencyOptions object
+ * @returns {String} - The cron definition for trigger
+ */
+export const fromFrequency = (frequency, options = {}) => {
+  const { dayOfMonth = 1, dayOfWeek = 1, hours = 0, minutes = 0 } = options
+  const sanitizedFrequency = VALID_FREQUENCIES.includes(frequency)
+    ? frequency
+    : DEFAULT_FREQUENCY
+
+  switch (sanitizedFrequency) {
+    case DAILY:
+      return `0 ${minutes} ${hours} * * *`
+    case HOURLY:
+      return `0 ${minutes} * * * *`
+    case MONTHLY:
+      return `0 ${minutes} ${hours} ${dayOfMonth} * *`
+    default:
+      // also WEEKLY
+      return `0 ${minutes} ${hours} * * ${dayOfWeek}`
+  }
+}
+
+/**
+ * Build a cron string for given konnector and from a given start date
+ *
+ * @param {import('../types').IOCozyKonnector} konnector - io.cozy.konnectors object
+ * @param {Date} startDate - start date
+ * @param {function} [randomDayTimeFn] - function generating random hours and minutes
+ * @returns {String} - The cron definition for trigger
+ */
+export const fromKonnector = (
+  konnector,
+  startDate = new Date(),
+  randomDayTimeFn = randomDayTime
+) =>
+  fromFrequency(konnector.frequency, {
+    ...randomDayTimeFn.apply(
+      null,
+      konnector.time_interval || DEFAULT_TIME_INTERVAL
+    ),
+    dayOfWeek: startDate.getDay(),
+    dayOfMonth: startDate.getDate()
+  })
```

Comment:
models are related to functional object. `cron` doesn't exist. `cron` is needed for the manifest, so let's move it there?